zebra-state/src/request.rs

```diff
@@ -439,12 +453,30 @@ pub enum ReadRequest {
     /// * [`Response::Transaction(None)`](Response::Transaction) otherwise.
     Transaction(transaction::Hash),
 
+    /// Looks up a Sapling note commitment tree either by a hash or height.
+    ///
+    /// Returns
+    ///
+    /// * [`ReadResponse::SaplingTree(Some(Arc<NoteCommitmentTree>))`](crate::ReadResponse::SaplingTree)
+    ///   if the corresponding block contains a Sapling note commitment tree.
+    /// * [`ReadResponse::SaplingTree(None)`](crate::ReadResponse::SaplingTree) otherwise.
+    SaplingTree(HashOrHeight),
+
     /// Looks up the balance of a set of transparent addresses.
     ///
     /// Returns an [`Amount`] with the total balance of the set of addresses.
     AddressBalance(HashSet<transparent::Address>),
 
-    /// Looks up transaction hashes that sent or received from addresses,
+    /// Looks up an Orchard note commitment tree either by a hash or height.
```

Comment:
Optional: Can you put the Sapling and Orchard requests together, before the Address index requests?

Reply:
Sure. I should have noticed.